### PR TITLE
sorter(ticdc): remove notifier from sorter merger

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: ref  #4351

### What is changed and how it works?
use a timer instead of notifier to reduce 1 go routine in sorter merger

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/sorter/unified/merger.go [hunk 1]
```diff
@@ -443,40 +441,46 @@ func runMerger(ctx context.Context, numSorters int, in <-chan *flushTask, out ch
...
+			curResolvedTs := atomic.LoadUint64(&minResolvedTs)
+			if curResolvedTs > lastResolvedTs {
+				err := onMinResolvedTsUpdate(curResolvedTs)
+				if err != nil {
+					return errors.Trace(err)
+				}
+			} else if curResolvedTs < lastResolvedTs {
+				log.Panic("resolved-ts regressed in sorter",
+					zap.Uint64("curResolved-ts", curResolvedTs),
+					zap.Uint64("lastResolved-ts", lastResolvedTs))
```
Comment (overvenus) [inline cdc/sorter/unified/merger.go:468]:
```suggestion
					zap.Uint64("curResolvedTs", curResolvedTs),
					zap.Uint64("lastResolvedTs", lastResolvedTs))
```

**Review by overvenus — APPROVED** [review]
LGTM

File: cdc/sorter/unified/merger.go [hunk 2]
```diff
@@ -443,40 +441,46 @@ func runMerger(ctx context.Context, numSorters int, in <-chan *flushTask, out ch
...
-		resolvedTsReceiver, err := resolvedTsNotifier.NewReceiver(time.Second * 1)
-		if err != nil {
-			if cerrors.ErrOperateOnClosedNotifier.Equal(err) {
-				// This won't happen unless `resolvedTsNotifier` has been closed, which is
-				// impossible at this point.
-				log.Panic("unexpected error", zap.Error(err))
-			}
-			return errors.Trace(err)
-		}
+		resolvedTsReceiver := time.NewTimer(time.Second * 1)
```
Comment (liuzix) [inline cdc/sorter/unified/merger.go]: Probably you should use a `Ticker` here

Comment (liuzix) [reply cdc/sorter/unified/merger.go]: You can call it `resolvedTsTicker`

**Review by liuzix — APPROVED** [review]

File: cdc/sorter/unified/merger.go (hunk 2 above)
Comment (sdojjy) [reply cdc/sorter/unified/merger.go]: fixed

**Pull request merged**